### ISSUE TITLE
Go Binaries in Distroless/Debian Don't Get Env Vars Injected

### ISSUE THREAD

**Issue by rejoshed** [bank-vaults/bank-vaults#1756]
**Describe the bug**:
When running a Go program directly from either a Distroless or Debian container the env vars are not injected.

**Expected behaviour**:
I expected the environment variables to be injected.

**Steps to reproduce the bug**:
Sample Go Program
``` Go
package main

import (
    "fmt"
    "log"
    "os"
)

func main() {
    //Set env a to b
    err := os.Setenv("a", "b")
    if err != nil {
        log.Fatal(err)
    }

    err = os.Setenv("c", "d")
    if err != nil {
        log.Fatal(err)
    }

    //Get all env variables
    fmt.Println(os.Environ())
    fmt.Println("*************************")
    fmt.Println("*************************")
    fmt.Println("*************************")
    fmt.Println("*************************")
    fmt.Println("*************************")

    fmt.Println(os.Getenv("AWS_INFRA_PLATFORM_NEXUS_ADMIN"))
}
```

A Dockerfile like this one will do. I was trying to get Crossplane to work with bank-vaults.
``` Dockerfile
FROM debian:latest

# Add upbound binary
COPY --from=xpkg.upbound.io/crossplane-contrib/provider-aws:v0.33.0 /usr/local/bin/crossplane-aws-provider /usr/local/bin/crossplane-aws-provider

RUN apt update -qqq
RUN apt install -yqqq ca-certificates
RUN update-ca-certificates

# Works when binary is not called directly.
ENTRYPOINT ["/bin/bash", "-c", "/usr/local/bin/crossplane-aws-provider --debug"]
```

**Additional context**:
If the binary is called directly vault-env will not replace the var.

If called through a shell as above in the entrypoint it will work. 

Basically something's up with vault-env, debian, and directly calling Go binaries.

/kind bug

**Comment by rejoshed** [comment]
So, after much back and forth, I've found that enabling daemon mode works as a workaround.

The issue must in general be to do with needed to be exec-ed or not.

When the vault-env process calls an intermediary or launches the go program without replacing itself with the go program the env vars work properly.


I do still see this as a bug, but at least I have a less ugly workaround for now.

**Comment by ramizpolic** [comment]
Thanks for sharing @ramizpolic, we will look into this
